custom/ilsgateway/models.py

```diff
@@ -136,7 +136,7 @@ class SupplyPointStatus(models.Model):
                                    max_length=50)
     status_value = models.CharField(max_length=50,
                                     choices=((c, c) for c in SupplyPointStatusValues.CHOICES))
-    status_date = models.DateTimeField(default=datetime.utcnow)
+    status_date = models.DateTimeField(default=datetime.utcnow, db_index=True)
     location_id = models.CharField(max_length=100, db_index=True)
```

Comment:
fwiw i don't think an integer foreign key would help that query. queries that have long lists in an IN clause are going to be slow and should be avoided. the alternative approach is to construct a temporary table with a single indexed column, insert the ids into the temp table, and join the temp table to the main result set.